web/modules/custom/dbcdk_community/src/Profile/ProfileRepository.php

```diff
@@ -0,0 +1,357 @@
+ *
+ * Related objects can be quarantines and community roles.
+ */
+class ProfileRepository {
+  use LoggerAwareTrait;
+
+  /**
+   * The API to use when retrieving profiles.
+   *
+   * @var \DBCDK\CommunityServices\Api\ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * The API to use when retrieving quarantines.
+   *
+   * @var \DBCDK\CommunityServices\Api\QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * The API to use when retrieving community roles.
+   *
+   * @var \DBCDK\CommunityServices\Api\CommunityRoleApi
+   */
+  protected $communityRoleApi;
+
+  /**
+   * ProfileRepository constructor.
+   *
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
+   *   The profile api to use.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The quarantine api to use.
+   * @param \DBCDK\CommunityServices\Api\CommunityRoleApi $community_role_api
+   *   The community role api to use.
+   */
+  public function __construct(ProfileApi $profile_api, QuarantineApi $quarantine_api, CommunityRoleApi $community_role_api) {
+    $this->profileApi = $profile_api;
+    $this->quarantineApi = $quarantine_api;
+    $this->communityRoleApi = $community_role_api;
+    $this->logger = new NullLogger();
+  }
+
+  /**
+   * Transform a filter data structure to the format expected by APIs.
+   *
+   * @param array $filter
+   *   The filter. This can contain entries such as where, limit, offset.
+   *
+   * @return string
+   *   A JSON encoded version of the filter.
+   *
+   * @see https://docs.strongloop.com/display/public/LB/Querying+data
+   */
+  protected function processFilter(array $filter) {
+    return (!empty($filter)) ? json_encode($filter) : NULL;
+  }
+
+  /**
+   * Retrieve a profile wih community roles and quarantines by id.
+   *
+   * @param int $id
+   *   The profile id to retrieve.
+   *
+   * @return Profile|NULL
+   *   The profile corresponding to the id. NULL if no profile is found.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   */
+  public function getProfile($id) {
+    $filter = [
+      'where' => ['id' => $id],
+    ];
+    return $this->findProfile($filter);
+  }
+
+  /**
+   * Retrieve a profile with community roles and quarantines by username.
+   *
+   * @param string $username
+   *   The username for the profile to retrieve.
+   *
+   * @return Profile|NULL
+   *   The profile corresponding to the username. NULL if no profile is found.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   */
+  public function getProfileByUsername($username) {
+    $filter = [
+      'where' => ['username' => $username],
+    ];
+    return $this->findProfile($filter);
+  }
+
+  /**
+   * Retrieve a profile matching a filter with community roles and quarantines.
```

Comment:
The basic functionality of this method is to: Find a profile.

That it contains CommunityRoles and Quarantines, shouldn't that be part of an expanded description? Example:

```
/**
 * Retrieve a profile matching a filter.
 *
 * The retrieved profile will be enriched with community roles and quarantines. We do this because...
...
```